src/runner.py

```diff
@@ -366,6 +367,9 @@ def _configure_runner(self) -> None:
         self.instance.execute(["/usr/bin/sudo", "chown", "ubuntu:ubuntu", str(self.runner_script)])
         self.instance.execute(["/usr/bin/sudo", "chmod", "u+x", str(self.runner_script)])
 
+        # Set permission to the same as GitHub-hosted runner for this directory.
+        self.instance.execute(["/usr/bin/sudo", "chmod", "777", "/usr/local/bin"])
```

Comment:
Should we have an end-to-end test that tries to interact with the folder to check that this permission is available to the ubuntu user?

Reply:
Ok, will be doing that.

Reply:
I don't think this is nice - you mentioned `777` being the default on the github runners, do you know why and where it comes from? I'd really explore going with `755`, it might be a small attack vector, though really anyone can place a runnable code inside /usr/local/bin that way and that directory is almost guaranteed to be on $PATH and it can take precedence over /usr/bin and /bin on some installs, meaning I can overwrite an existing command with whatever I want to.

Reply:
Keep in mind that this machine is essentially designed to allow the GitHub workflow to do anything, they can already sudo and mess with the entire machine. The machine gets thrown away and a new clean machine created for the next workflow run

Reply:
However, GitHub Action such as [SkyWalking Eyes](https://github.com/apache/skywalking-eyes) (used in operator-workflow for checking licences) requires write permission to /usr/local/bin as a user.
From testing with a GitHub-hosted runner, the /usr/local/bin does have `drwxrwxrwx`.
```
2023-05-16T06:26:52.8224628Z ##[group]Run ls -la /usr/local
2023-05-16T06:26:52.8224975Z  [36;1mls -la /usr/local [0m
2023-05-16T06:26:52.8281068Z shell: /usr/bin/bash -e {0}
2023-05-16T06:26:52.8281367Z ##[endgroup]
2023-05-16T06:26:52.8565654Z total 92
2023-05-16T06:26:52.8567120Z drwxr-xr-x 19 root root  4096 May  7 22:16 .
2023-05-16T06:26:52.8568273Z drwxr-xr-x 14 root root  4096 Apr 28 02:09 ..
2023-05-16T06:26:52.8569131Z drwxrwxrwx 10 root root  4096 May  7 21:49 .ghcup
2023-05-16T06:26:52.8570111Z drwxr-xr-x  3 root root  4096 May  7 21:27 aws-cli
2023-05-16T06:26:52.8570973Z drwxr-xr-x  3 root root  4096 May  7 21:27 aws-sam-cli
2023-05-16T06:26:52.8571891Z drwxrwxrwx  2 root root  4096 May  7 22:37 bin
2023-05-16T06:26:52.8572634Z drwxr-xr-x  3 root root  4096 Apr  4 19:10 doc
2023-05-16T06:26:52.8573722Z drwxr-xr-x  2 root root  4096 Apr 28 02:09 etc
2023-05-16T06:26:52.8574621Z drwxr-xr-x  2 root root  4096 Apr 28 02:09 games
2023-05-16T06:26:52.8575704Z drwxr-xr-x  3 root root  4096 May  7 21:58 include
2023-05-16T06:26:52.8576553Z drwxr-xr-x  8 root root  4096 May  7 22:16 julia1.8.5
2023-05-16T06:26:52.8577717Z drwxr-xr-x  8 root root  4096 May  7 22:18 lib
2023-05-16T06:26:52.8578477Z drwxr-xr-x  4 root root  4096 Apr  4 19:10 man
2023-05-16T06:26:52.8579463Z drwxr-xr-x  3 root root  4096 May  7 21:58 n
2023-05-16T06:26:52.8581003Z drwxr-xr-x  2 root root  4096 May  7 21:11 sbin
2023-05-16T06:26:52.8582187Z drwxr-xr-x  3 root root  4096 May  7 21:27 sessionmanagerplugin
2023-05-16T06:26:52.8583055Z drwxr-xr-x 23 root root  4096 May  7 22:40 share
2023-05-16T06:26:52.8584201Z drwxr-xr-x  2 root root 20480 May  7 21:57 sqlpackage
2023-05-16T06:26:52.8585631Z drwxr-xr-x  2 root root  4096 Apr 28 02:09 src
```

Reply:
While the GitHub Actions are run under a user, the user are given sudo access within the GitHub-hosted runner and self-hosted runner.
Any GitHub Actions could already overwrite binaries in /usr/local/bin or any other file.